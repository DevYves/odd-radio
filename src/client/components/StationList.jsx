import React, {Component} from 'react';
import Station from './Station.jsx';

class StationList extends Component {
  render() {
    const station = this.props.stations.map((post, index) => {
      if (index % 2 === 0) {
        return <Station
          key={ post.id }
          id={ post.id }
          name={ post.name }
          description={ post.description }
          audioFeed={ post.audio_feed }
          type={ post.stream_type }
          homePage={ post.home_page }
          city={ post.city }
          handleSelectedStation={ this.props.handleSelectedStation }
          stationType={1}
           />
      } else {
        return <Station
          key={ post.id }
          id={ post.id }
          name={ post.name }
          description={ post.description }
          audioFeed={ post.audio_feed }
          type={ post.stream_type }
          homePage={ post.home_page }
          city={ post.city }
          handleSelectedStation={ this.props.handleSelectedStation }
          stationType={2}
           />
      }
    });

    return (
      <section className="station-list-container">
        { station }
      </section>
    )
  }
}
export default StationList;