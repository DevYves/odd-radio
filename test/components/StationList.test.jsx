import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

import StationList from '../../src/client/components/StationList';

describe('StationList', () => {
  const props = {
    handleSelectedStation: jest.fn(),
    stations: [{
        id: 1,
        name: 'CFCR',
        audio_feed: 'http://onair.cfcr.ca/hifi.mp3',
        home_page: 'http://cfcr.ca/',
        city: 'Saskatoon',
        province: 'SK',
        description: 'CFCR 90.5 FM Community Radio is a non-profit corporation owned by the Community Radio Society of Saskatoon. We are a volunteer powered, listener supported organization serving the alternative radio needs of Saskatoon and surrounding areas.',
        frequency: '90.5 FM',
        language: 'English',
        stream_type: 'audio/mp3'
    },
       {
        id: 2,
        name:'CFMU' ,
        audio_feed: 'http://138.197.136.105:8000/mount.128mp3' ,
        home_page:'http://cfmu.ca/' ,
        city: 'Hamilton',
        province:'ON',
        description:'93.3 CFMU-FM, Canadaâ€™s third oldest campus / community radio station, has been redefining radio for over 25 years, delivering daily the most enlightening, innovative and entertaining programming in our broadcast area.',
        frequency: '93.3 FM',
        language:'English' ,
        stream_type: 'audio/mp3'
      }]
  }

  const subject = <StationList { ...props }/>;

  it('matches snapshot', () => {
    expect(renderer.create(subject)).toMatchSnapshot();
  });


});