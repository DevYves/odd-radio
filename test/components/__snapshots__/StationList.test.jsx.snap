// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StationList matches snapshot 1`] = `
<section
  className="station-list-container"
>
  <div
    id="CFCR"
  >
    <div
      className="container station-container1"
    >
      <div
        className="row station-row border"
      >
        <div
          className="three columns station-branding center"
        >
          <div
            className="non-expanded-station-name center"
            onClick={[Function]}
          >
             
             
            CFCR
          </div>
        </div>
        <div
          className="three columns station-play-button center"
        >
          <div>
            <i
              className="fa fa-pause fa-2x station-play-button"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="three columns station-info-button center"
        >
          <i
            aria-hidden="true"
            className="fa fa-chevron-down card-chevron fa-2x"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
  <div
    id="CFMU"
  >
    <div
      className="container station-container2"
    >
      <div
        className="row station-row border"
      >
        <div
          className="three columns station-branding center"
        >
          <div
            className="non-expanded-station-name center"
            onClick={[Function]}
          >
             
             
            CFMU
          </div>
        </div>
        <div
          className="three columns station-play-button center"
        >
          <div>
            <i
              className="fa fa-play fa-2x station-play-button"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="three columns station-info-button center"
        >
          <i
            aria-hidden="true"
            className="fa fa-chevron-down card-chevron fa-2x"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="container masthead"
    >
      <div
        className="row station-row border"
      >
        <div
          className="one-third column center"
          onClick={[Function]}
        >
          About
        </div>
        <div
          className="one-third column center"
          onClick={[Function]}
        >
          Contact
        </div>
        <div
          className="one-third column center"
          onClick={[Function]}
        >
          Team
        </div>
      </div>
      <div
        className="row border"
      >
        <div
          className="column"
        />
      </div>
    </div>
  </div>
</section>
`;
